### PR TITLE
feat: saliency maps

### PR DESCRIPTION
**Pull request by vivek-mandal**
Added Generalised model
Updated the notebook link with new EVA API

### CODE REVIEW

**Comment by jarulraj** [comment]
Is it possible to visualize the features on the image in the notebook? remove the old API notebook

File: eva/udfs/saliency_feature_extractor.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import torch.nn as nn
+import torch.nn.functional as F
+
+from eva.catalog.catalog_type import NdArrayType
+from eva.udfs.abstract.abstract_udf import AbstractUDF
+from eva.udfs.decorators.decorators import forward, setup
+from eva.udfs.decorators.io_descriptors.data_types import PandasDataframe
+from eva.udfs.gpu_compatible import GPUCompatible
+from torchvision.transforms import Compose, ToTensor, Resize
+from PIL import Image
```
Comment (affan00733) [inline eva/udfs/saliency_feature_extractor.py:30]: make sure you had successfully executed `bash script/test/test.sh -m "LINTER"   `

**Review by affan00733 — CHANGES_REQUESTED** [review]

**Comment by jarulraj** [comment]
Running this test on a server with GPU:

`pytest -k 'test_saliency' -s -v  `

```eva.executor.executor_utils.ExecutorError: Input type (torch.FloatTensor) and weight type (torch.cuda.FloatTensor) should be the same or input should be a MKLDNN tensor and weight is a dense tensor```

**Comment by jarulraj** [comment]
@vivek-mandal Look at my changes in -- https://github.com/georgia-tech-db/eva/tree/vivek-mandal-SaliencyMaps_Updated

**Pull request merged**